File: dotnet/CucumberExpressions/Expression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

namespace CucumberExpressions;

public interface Expression {
    //List<Argument<?>> match(String text, Type... typeHints);

    Regex getRegexp();

    String getSource();
}

File: java/src/main/java/io/cucumber/cucumberexpressions/Expression.java
package io.cucumber.cucumberexpressions;

import org.apiguardian.api.API;

import java.lang.reflect.Type;
import java.util.List;
import java.util.Optional;
import java.util.regex.Pattern;

@API(status = API.Status.STABLE)
public interface Expression {

    /**
     * Matches a string to an expression. Empty if no match.
     */
    Optional<List<Argument<?>>> match(String text, Type... typeHints);

    Pattern getRegexp();

    String getSource();
}
